app/src/main/java/mozilla/lockbox/presenter/OnboardingAutofillPresenter.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.presenter
+
+import android.annotation.TargetApi
+import android.os.Build
+import androidx.annotation.RequiresApi
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.addTo
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import mozilla.lockbox.action.OnboardingStatusAction
+import mozilla.lockbox.action.RouteAction
+import mozilla.lockbox.action.SettingAction
+import mozilla.lockbox.action.SettingIntent
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.flux.Presenter
+
+interface OnboardingAutofillView {
+    val onDismiss: Observable<Unit>
+    val onEnable: Observable<Unit>
+}
+
+@ExperimentalCoroutinesApi
+class OnboardingAutofillPresenter(
+    private val view: OnboardingAutofillView,
+    private val dispatcher: Dispatcher = Dispatcher.shared
+) : Presenter() {
+
+    @TargetApi(Build.VERSION_CODES.O)
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onViewReady() {
+
+        view.onDismiss.subscribe {
+            dispatcher.dispatch(OnboardingStatusAction(false))
+        }?.addTo(compositeDisposable)
+
+        view.onEnable.subscribe {
+            dispatcher.dispatch(SettingAction.Autofill(true))
```

Comment:
ok, interesting (and non-obvious! again! I'm sorry D: ) thing here, the autofill setting is actually driven by the system setting (whether Lockbox is selected as a credential provider or nowt). As of right now, dispatching `SettingAction.Autofill(true)` won't have any affect on system because the user needs to go to their system settings to toggle the autofill setting. That's not to say you can't do it, just letting you know that it won't do anything!!

Reply:
ok ok I see what you're saying :)